src/uwtools/drivers/sfc_climo_gen.py

```diff
@@ -0,0 +1,162 @@
+"""
+
+import os
+import stat
+from pathlib import Path
+from typing import Any, Dict
+
+from iotaa import asset, dryrun, task, tasks
+
+from uwtools.config.formats.nml import NMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.utils.processing import execute
+from uwtools.utils.tasks import file
+
+
+class SfcClimoGen(Driver):
+    """
+    A driver for sfc_climo_gen.
+    """
+
+    def __init__(self, config_file: Path, dry_run: bool = False, batch: bool = False):
+        """
+        The driver.
+
+        :param config_file: Path to config file.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config_file=config_file, dry_run=dry_run, batch=batch)
+        if self._dry_run:
+            dryrun()
+        self._rundir = Path(self._driver_config["run_dir"])
+
+    # Workflow tasks
+
+    @task
+    def namelist_file(self):
+        """
+        The namelist file.
+        """
+        fn = "fort.41"
+        yield self._taskname(f"namelist file {fn}")
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        vals = self._driver_config["namelist"]["update_values"]["config"]
+        input_paths = [Path(v) for k, v in vals.items() if k.startswith("input_")]
+        input_paths += [Path(vals["mosaic_file_mdl"])]
+        input_paths += [Path(vals["orog_dir_mdl"]) / fn for fn in vals["orog_files_mdl"]]
+        yield [file(input_path) for input_path in input_paths]
+        self._create_user_updated_config(
+            config_class=NMLConfig,
+            config_values=self._driver_config.get("namelist", {}),
+            path=path,
+        )
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.namelist_file(),
+            self.runscript(),
+        ]
+
+    @tasks
+    def run(self):
+        """
+        Run execution.
+        """
+        yield self._taskname("run")
+        yield (self._run_via_batch_submission() if self._batch else self._run_via_local_execution())
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        envcmds = self._driver_config.get("execution", {}).get("envcmds", [])
+        execution = [self._runcmd, "test $? -eq 0 && touch %s/done" % self._rundir]
+        scheduler = self._scheduler if self._batch else None
+        path.parent.mkdir(parents=True, exist_ok=True)
+        rs = self._runscript(envcmds=envcmds, execution=execution, scheduler=scheduler)
+        with open(path, "w", encoding="utf-8") as f:
+            print(rs, file=f)
+        os.chmod(path, os.stat(path).st_mode | stat.S_IEXEC)
```

Comment:
We'll probably want to do even more DRY refactoring once we have a third driver and see yet more commonalities. For example, while the arguments to `self._runscript()` vary between this driver and the `fv3` one, lines 89-93 will probably be mostly identical between drivers and could be factored out to a common method, with the arguments to `self._runscript()` being passed to that common method. In particular, the `chmod` command to make the script executable is ugly/weird and should probably not be repeatedly duplicated.

I propose that, if we don't find unexpected unique behavior in our third driver, that we should do a refactoring sweep and see what can be factored out either to the `uwtools.drivers.driver.Driver` class or to the `uwtools.utils.tasks` module.